### DISCUSSION TITLE
Dynamic whereclause causes: Array index [4] out of range, array size is [3]

### DISCUSSION THREAD

**Discussion by MvdO79** [wheels-dev/wheels#1271] [Q&A] [answered]
Hi all!

Using this:

`listings=model("listing").findAll(where="#whereQuery#", page=params.page, perPage=6, order="pk_listings DESC", handle="listingsData");`

and this query in #whereQuery#:

`status='1' AND headcategory='2' AND (country='Nederland' OR fk_listings_users IN(SELECT pk_users FROM users WHERE country='Nederland'))`

i get an error:

`<html>
<body>

Array index [4] out of range, array size is [3]
--
Stacktrace | The Error Occurred inD:\odrive\OneDrive\Websites\advertenties\wheels\model\sql.cfm: line 489487: }488: if (local.i < ListLen(local.where, "?")) {489: local.column = local.params[local.i].column;490: ArrayAppend(local.rv, local.column & " " & local.params[local.i].operator);491: local.param = {type=local.params[local.i].type, dataType=local.params[local.i].dataType, scale=local.params[local.i].scale, list=local.params[local.i].list};

</body>
</html>`

Any ideas?

**Comment by edthetechie** [comment, marked as answer]
You can't use a nested query in a where clause. Create it as a property on
the model and then reference that variable in your where clause.

Hope that helps
